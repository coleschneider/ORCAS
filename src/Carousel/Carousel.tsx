import * as React from 'react';
import SwipeableViews from 'react-swipeable-views';
import Pagination from './Pagination';
const styles = {
  slide: {
    padding: 15,
    minHeight: 100,
    color: '#fff',
  },
  slide1: {
    backgroundColor: '#FEA900',
  },
  slide2: {
    backgroundColor: '#B3DC4A',
  },
  slide3: {
    backgroundColor: '#6AC0FF',
  },
};

interface CarouselProps {
  children: React.ReactChildren;
}
interface CarouselState {
  index: number;
}
class Carousel extends React.Component<CarouselProps, CarouselState> {
  static Slide = ({ children, author, avatar }) => (
    <div className="meet__slide">
      <img alt="Author image" src={avatar} className="meet__avatar" />
      <p>{children}</p>
      <div className="meet__author">
        <span className="meet__name">{author}</span>
        <a href="#0" className="meet__link">
          @{author}
        </a>
      </div>
    </div>
  );
  constructor(props) {
    super(props);

    this.state = { index: 0 };
    this.handleChangeIndex = this.handleChangeIndex.bind(this);
  }

  handleChangeIndex(index) {
    this.setState({ index });
  }

  render() {
    const { children } = this.props;
    const total = React.Children.count(children);
    return (
      <div className="meet-wrap">
        <div className="row">
          <div className="col-full meet-header">
            <h2 className="display-2">Meet the team!</h2>
          </div>
        </div>
        <div className="row meet slick-slider">
          <SwipeableViews
            index={this.state.index}
            onChangeIndex={this.handleChangeIndex}
            enableMouseEvents={true}
            resistance={true}
          >
            {this.props.children}
          </SwipeableViews>
          <Pagination total={total} index={this.state.index} onChangeIndex={this.handleChangeIndex} />
        </div>
      </div>
    );
  }
}
export default Carousel;
