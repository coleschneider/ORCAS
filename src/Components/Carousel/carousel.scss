@import '../../stylesheets/layout/colors';

/* Next & previous buttons */
.prev, .next {
  cursor: pointer;
  position: absolute;
  top: 50%;
  width: auto;
  margin-top: -30px;
  display: flex;
  // padding: 16px;
  color: #888;
  font-weight: bold;
  font-size: 2em;
  border-radius: 0 3px 3px 0;
  user-select: none;
}

/* Position the "next button" to the right */
.next {
  position: absolute;
  right: 0;
  border-radius: 3px 0 0 3px;
}

/* On hover, add a black background color with a little bit see-through */
.prev:hover, .next:hover {
  
  color: darken($color: #888, $amount: 80%)
}

/* ------------------------------------------------------------------- 
 * ## slick slider
 * ------------------------------------------------------------------- */
 .slick-slider {
  position: relative;
}

.slick-slider .slick-slide {
  outline: none;
}

.slick-slider .slick-dots {
  display: block;
  list-style: none;
  width: 100%;
  padding: 0;
  margin: 4.2rem 0 0 0;
  text-align: center;
  position: absolute;
  top: 100%;
  left: 0;
}

.slick-slider .slick-dots li {
  display: inline-block;
  width: 27px;
  height: 27px;
  margin: 0;
  padding: 9px;
  cursor: pointer;
}

.slick-slider .slick-dots li button {
  display: block;
  width: 10px;
  height: 10px;
  line-height: 10px;
  border-radius: 50%;
  background: rgba(0, 0, 0, 0.4);
  border: none;
  padding: 0;
  margin: 0;
  cursor: pointer;
  font: 0/0 a;
  text-shadow: none;
  color: transparent;
}

.slick-slider .slick-dots li button:hover,
.slick-slider .slick-dots li button:focus {
  outline: none;
}

.slick-slider .slick-dots li.slick-active button,
.slick-slider .slick-dots li:hover button {
  background: #80d3d2;
}






/* ===================================================================
 * # meet
 *
 * ------------------------------------------------------------------- */

 .meet-header {
  text-align: center;
  padding-top: 6.4rem;
  position: relative;
}

.meet-header::before {
  content: "";
  display: inline-block;
  height: 1px;
  width: 60%;
  background-color: $grey;
  position: absolute;
  top: 0;
  left: 50%;
  -webkit-transform: translate3d(-50%, 0, 0);
  -ms-transform: translate3d(-50%, 0, 0);
  transform: translate3d(-50%, 0, 0);
}

.meet {
  font-family: "Nunito Sans", sans-serif;
  font-weight: normal;
  font-size: 1.7rem;
  line-height: 1.882;
  margin-top: 4rem;
  padding-bottom: 7.2rem;
  position: relative;
}

.meet__slide {
  position: relative;
  padding: 16px;
  z-index: 500;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  margin: 0px auto;
  width: 50%;
}
.senior-card {
  
}
.meet__avatar {
  width: 7.2rem;
  height: 7.2rem;
  border-radius: 100%;
  margin-right: 8px;
  outline: none;
  position: relative;
  float: left;
  top: 0;
  left: 0;
}

.meet__author {
  display: inline-block;
  position: relative;
  margin-bottom: 1.6rem;
  line-height: 2.4rem;
}

.meet__name,
.meet__link {
  display: inline-block;
}

.meet__name {
  font-weight: 700;
  color: #000000;
  margin-right: 4px;
}

.meet__link {
  font-size: 1.6rem;
  color: rgba(0, 0, 0, 0.4);
}


/* ------------------------------------------------------------------- 
 * responsive:
 * meet
 * ------------------------------------------------------------------- */


@media only screen and (max-width:600px) {
  .meet {
    margin-top: 3.2rem;
  }

  .meet-header {
    padding-top: 6.4rem !important;
  }
  .prev, .next {
    display: none;
  }
  
  .meet__slide {
    width: 80%;
    
  }

}
.quote {
    font-size: 80px;
    line-height: 0.6em;
    opacity: 0.2;
    display: flex;
}
.quote-2 {
    opacity: 0.6;
 
}