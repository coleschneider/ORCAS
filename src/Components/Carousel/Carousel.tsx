import * as React from 'react';
import SwipeableViews from 'react-swipeable-views';
import Pagination, { NextPage, PrevPage } from './Pagination/Pagination';
import './carousel.scss';
const styles = {
  slide: {
    color: '#fff',
    minHeight: 100,
    padding: 15,
  },
  slide1: {
    backgroundColor: '#FEA900',
  },
  slide2: {
    backgroundColor: '#B3DC4A',
  },
  slide3: {
    backgroundColor: '#6AC0FF',
  },
};

interface CarouselProps {
  children: React.ReactChildren;
}
interface CarouselState {
  index: number;
}
class Carousel extends React.Component<CarouselProps, CarouselState> {
  static Slide = ({ children, author, avatar }) => (
    <div className="meet__slide">
      <img alt="Author image" src={avatar} className="meet__avatar" />
      <span className="quote">‚ùù</span>
      <p>{children}</p>
      <div className="meet__author">
        <span className="meet__name">{author}</span>
        <a href="#0" className="meet__link">
          @{author}
        </a>
      </div>
    </div>
  );
  constructor(props) {
    super(props);

    this.state = { index: 0 };
    this.handleChangeIndex = this.handleChangeIndex.bind(this);
  }

  handleChangeIndex(nextIndex) {
    const { children } = this.props;
    const total = React.Children.count(children);
    this.setState({
      index: nextIndex > total - 1 ? 0 : nextIndex < 0 ? total - 1 : nextIndex,
    });
  }

  render() {
    const { children } = this.props;
    const total = React.Children.count(children);
    const { index } = this.state;
    return (
      <div className="row meet slick-slider">
        <SwipeableViews
          // style={{overflow: 'unset'}}
          // slideStyle={{overflow: 'visible'}}
          index={index}
          slideClassName="senior-card"
          onChangeIndex={this.handleChangeIndex}
          enableMouseEvents={true}
          resistance={true}
        >
          {this.props.children}
        </SwipeableViews>
        <PrevPage onClick={() => this.handleChangeIndex(index - 1)} />
        <NextPage onClick={() => this.handleChangeIndex(index + 1)} />
        <Pagination total={total} index={index} onChangeIndex={this.handleChangeIndex} />
      </div>
    );
  }
}
export default Carousel;
