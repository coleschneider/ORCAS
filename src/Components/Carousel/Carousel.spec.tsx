import * as React from 'react';
import Carousel from './Carousel';

describe('Carousel', () => {
  let children = [];
  let wrapper;
  const setupViews = count => {
    for (let i = 0; i < count; i++) {
      children.push(<div key={i} />);
    }
    wrapper = mount(<Carousel>{children.map(c => c)}</Carousel>);
  };
  beforeEach(() => (children = []));
  it('should match the snapshot', () => {
    setupViews(1);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render the total pages from children count', () => {
    setupViews(3);
    expect(wrapper.find('Pagination').props().total).toBe(3);
  });
  it('should increment the number if it is less than total and greater than 0', () => {
    setupViews(3);
    wrapper.instance().handleChangeIndex(1);
    expect(wrapper.state().index).toBe(1);
  });
  it('should start at 0 if the nextIndex is greater than total', () => {
    setupViews(2);
    wrapper.instance().handleChangeIndex(2);
    expect(wrapper.state().index).toBe(0);
  });
  it('should start at total if the nextIndex is less than 0', () => {
    setupViews(2);
    wrapper.instance().handleChangeIndex(-1);
    expect(wrapper.state().index).toBe(1);
  });
});
