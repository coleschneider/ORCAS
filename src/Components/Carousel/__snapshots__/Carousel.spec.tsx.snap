// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel should match the snapshot 1`] = `
<div
  className="meet-wrap"
>
  <div
    className="row"
  >
    <div
      className="col-full meet-header"
    >
      <h2
        className="display-2"
      >
        Meet the team!
      </h2>
    </div>
  </div>
  <div
    className="row meet slick-slider"
  >
    <ReactSwipableView
      animateHeight={false}
      animateTransitions={true}
      axis="x"
      disableLazyLoading={false}
      disabled={false}
      enableMouseEvents={true}
      hysteresis={0.6}
      ignoreNativeScroll={false}
      index={0}
      onChangeIndex={[Function]}
      resistance={true}
      springConfig={
        Object {
          "delay": "0s",
          "duration": "0.35s",
          "easeFunction": "cubic-bezier(0.15, 0.3, 0.25, 1)",
        }
      }
      threshold={5}
    >
      child
    </ReactSwipableView>
    <Pagination
      index={0}
      onChangeIndex={[Function]}
      total={1}
    />
  </div>
</div>
`;
